suite/tests/api/it_arm.c

```diff
@@ -0,0 +1,123 @@
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "dr_api.h"
+
+#define ASSERT(x) \
+    ((void)((!(x)) ? \
+        (dr_fprintf(STDERR, "ASSERT FAILURE: %s:%d: %s\n", __FILE__,  __LINE__, #x),\
+         dr_abort(), 0) : 0))
+
+void
+test_dr_insert_it_instrs_cbr(void *dcontext)
+{
+    instrlist_t *ilist = instrlist_create(dcontext);
+    instr_t *where = INSTR_CREATE_label(dcontext);
+
+    instr_t *instr_it1, *instr_it2, *instr_it3;
+    byte buffer[4096];
+
+    instrlist_append(ilist, where);
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_jump
+            (dcontext,
+             opnd_create_instr(where)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_jump
+            (dcontext,
+             opnd_create_instr(where)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_move
+            (dcontext,
+             opnd_create_reg(DR_REG_R1),
+             opnd_create_reg(DR_REG_R2)));
+    instrlist_preinsert(ilist, where, XINST_CREATE_jump
```

Comment:
Please add a test of a predicated call instruction too which is what #1702 was originally about